Purchases/Public/RCPurchases.m

```diff
@@ -721,23 +722,93 @@ - (void)invalidatePurchaserInfoCache {
 #pragma mark Subcriber Attributes
 
 - (void)setAttributes:(NSDictionary<NSString *, NSString *> *)attributes {
-    [self _setAttributes:attributes];
+    RCDebugLog(@"setAttributes called");
+    [self.subscriberAttributesManager setAttributes:attributes appUserID:self.appUserID];
 }
 
 - (void)setEmail:(nullable NSString *)email {
-    [self _setEmail:email];
+    RCDebugLog(@"setEmail called");
+    [self.subscriberAttributesManager setEmail:email appUserID:self.appUserID];
 }
 
 - (void)setPhoneNumber:(nullable NSString *)phoneNumber {
-    [self _setPhoneNumber:phoneNumber];
+    RCDebugLog(@"setPhoneNumber called");
+    [self.subscriberAttributesManager setPhoneNumber:phoneNumber appUserID:self.appUserID];
 }
 
 - (void)setDisplayName:(nullable NSString *)displayName {
-    [self _setDisplayName:displayName];
+    RCDebugLog(@"setDisplayName called");
+    [self.subscriberAttributesManager setDisplayName:displayName appUserID:self.appUserID];
 }
 
 - (void)setPushToken:(nullable NSData *)pushToken {
-    [self _setPushToken:pushToken];
+    RCDebugLog(@"setPushToken called");
+    [self.subscriberAttributesManager setPushToken:pushToken appUserID:self.appUserID];
```

Comment:
we might be able to improve these with an enum, but it's out of scope for this PR